### ISSUE TITLE
Placing the cursor beyond the end of a line should not extend the line until a char is typed

### ISSUE THREAD

**Issue by JohnScholes** [Dyalog/ride#394]
### Describe the issue you are having
Just placing the cursor beyond the end of a line shouldn't extend the line. The line should be padded to  the cursor position _only_ on receipt of a typed character.

cf issues: #238 and #381

### How do you reproduce the issue?
```
      ⎕←text←'singing' 'the' 'blues'
┌───────┬───┬─────┐
│singing│the│blues│
└───────┴───┴─────┘
      )ed text
```
Click to place the cursor beyond the end of the first line,
Cursor Down,
Cursor back Up again,
Escape to save (although no chars typed) and quit.
```
      text
┌──────────┬──────────┬─────┐
│singing   │the       │blues│
└──────────┴──────────┴─────┘
```
Notice that both the first and second items have been (wrongly) extended with blanks.
### Paste the contents of Help → About (Shift+F1)
```
IDE:
  Version: 4.1.3258
  Platform: MacIntel
  Date: 2018-05-09 21:37:18 +0200
  Git commit: a2dd832e5712e4dcf8d7f69d4f7e3a561d5123f4
  Preferences:{
    "autocompletion":"shell",
    "breakPts":"0",
    "colourScheme":"Zen",
    "colourSchemes":"[{\"name\":\"Zen\",\"theme\":\"light\",\"styles\":\"asgn=fg:#000000 com=fg:#000000 dfn=fg:00f diam=fg:#000000 err=fg:f00 fn=fg:#000000 idm=fg:#000000 kw=fg:#000000 lnum=fg:#000000,bg:f,bgo:0 mod=bg:7,bgo:0.25 mtch=bg:ff8,bgo:0.5 norm=bg:f,bgo:1 ns=fg:#000000 num=fg:#000000 op1=fg:#000000 op2=fg:#000000 par=fg:#000000 quad=fg:#000000 qdl=fg:#000000 sel=bg:48e,bgo:0.5 semi=fg:#000000 sqbr=fg:#000000 srch=bg:f80,bgo:0.5 str=fg:#000000 tc=bg:#ffffff,bgo:1,fg:0 tcpe=bg:c8c8c8,bgo:1 trad=fg:8 var=fg:#000000 zld=fg:#000000 scmd=fg:#000000 ucmd=fg:#000000 vtt=bg:ff0 ca=bg:#ffffff,bgo:1,fg:#000000 cm=bg:#ffffff,bgo:0.1,fg:#000000 cv=bg:f,bgo:1,fg:0 cvv=bg:#ffffff,bgo:1,fg:#000000 ma=bg:#ffffff,bgo:1,fg:#000000 na=bg:#ffffff,bgo:1,fg:#000000 qor=bg:#ffffff,bgo:1,fg:#000000 dfn1=fg:0 dfn2=fg:0 dfn3=fg:0 dfn4=fg:0 dfn5=fg:0\"}]",
    "cursorBeyondEOL":"1",
    "editWins":"{\"width\":1092,\"height\":585,\"x\":345,\"y\":23,\"ox\":0,\"oy\":0}",
    "editWinsRememberPos":"1",
    "floatOnTop":"1",
    "floating":"1",
    "fold":"0",
    "indentOnOpen":"0",
    "kbdLocale":"en_GB_Mac",
    "lbar":"0",
    "lineNums":"0",
    "minimapRenderCharacters":"0",
    "showEditorToolbar":"0",
    "sqp":"0",
    "valueTips":"0",
    "wrap":"1",
    "zoom":"5"
  }

Interpreter:
  Version: 17.0.33027
  Platform: Mac-64
  Edition: Unicode/64
  Date: May  4 2018 at 01:56:04
```

**Comment by JohnScholes** [comment]
Historical note: APL input is based on Quote-Quad input. 
- A line is presented on the terminal and the input cursor  placed at a given position.
- The cursor may be moved  without effecting the input
- Any characters typed cause blank-padding, if necessary, to that position.
- On hitting Enter, the cursor position is forgotten but the whole line is returned as the result of the input.

This differs from the, these days, more common approach of having an immutable "prompt" field followed by an input field. Viewed this way "six-space prompt" is a misnomer. Six blanks are not written to the terminal as a prompt; instead, the cursor is positioned in the 7th column, and a read issued.

**Comment by e9gille** [comment]
Yup, it shouldn't. Unfortunately I can't place the cursor in a virtual position with monaco, so I had to pad the lines without any input.
I'll need to follow up with a cleanup job of undoing the padding if line left untouched.

**Comment by e9gille** [comment]
I've looked into this again and it has proven to be more complicated than I first anticipated. If I only had to worry about a single cursor I think I could fix the remaining issues without too much effort. As the monaco editor supports multiple cursors I would have to cover all the scenarios of using those as well, which is non-trivial.

I'm afraid I'll need to revert the support for this for now and maybe revisit some other time.
